src/app/pages/sharing/webdav/webdav-form/webdav-form.component.ts

```diff
@@ -1,112 +1,137 @@
-          placeholder: helptextSharingWebdav.placeholder_path,
-          tooltip: helptextSharingWebdav.tooltip_path,
-          required: true,
-          validation: helptextSharingWebdav.validator_path,
-        },
-        {
-          type: 'checkbox',
-          name: 'ro',
-          placeholder: helptextSharingWebdav.placeholder_ro,
-          tooltip: helptextSharingWebdav.tooltip_ro,
-        },
-        {
-          type: 'checkbox',
-          name: 'perm',
-          value: true,
-          placeholder: helptextSharingWebdav.placeholder_perm,
-          tooltip: helptextSharingWebdav.tooltip_perm,
-        },
-        {
-          type: 'checkbox',
-          name: 'enabled',
-          value: true,
-          placeholder: helptextSharingWebdav.placeholder_enabled,
-          tooltip: helptextSharingWebdav.tooltip_enabled,
-        },
-      ],
-    }];
+  readonly treeNodeProvider = this.filesystemService.getFilesystemNodeProvider();
 
   constructor(
-    protected router: Router,
+    private fb: FormBuilder,
     protected ws: WebSocketService,
+    private translate: TranslateService,
+    private slideInService: IxSlideInService,
+    private cdr: ChangeDetectorRef,
     private dialog: DialogService,
+    private errorHandler: FormErrorHandlerService,
     private loader: AppLoaderService,
-    private translate: TranslateService,
-  ) {}
-
-  afterInit(entityForm: EntityFormComponent): void {
-    entityForm.formGroup.controls['perm'].valueChanges.pipe(untilDestroyed(this)).subscribe((value: boolean) => {
+    private filesystemService: FilesystemService,
+  ) {
+    this.form.controls.perm.valueChanges.pipe(untilDestroyed(this)).subscribe((value: boolean) => {
       this.confirmSubmit = value;
     });
-    this.title = entityForm.isNew ? this.translate.instant('Add WebDAV') : this.translate.instant('Edit WebDAV');
   }
 
-  afterSave(): void {
+  setWebdavForEdit(webdav: WebDavShare): void {
+    this.editingWebdav = webdav;
+    this.form.patchValue({
+      ...webdav,
+    });
+  }
+
+  onSubmit(): void {
+    if (this.confirmSubmit) {
+      this.dialog.confirm({
+        title: helptextSharingWebdav.warning_dialog_title,
+        message: helptextSharingWebdav.warning_dialog_message,
+        hideCheckBox: false,
+      }).pipe(filter(Boolean), untilDestroyed(this)).subscribe(() => {
+        this.saveConfig();
+      });
+    } else {
+      this.saveConfig();
+    }
+  }
+
+  saveConfig(): void {
+    const values = {
+      ...this.form.value,
+    };
+
+    this.isFormLoading = true;
+    let request$: Observable<unknown>;
+    if (this.editingWebdav) {
+      request$ = this.ws.call('sharing.webdav.update', [
+        this.editingWebdav.id,
+        values,
+      ]);
+    } else {
+      request$ = this.ws.call('sharing.webdav.create', [values]);
+    }
+
+    request$.pipe(untilDestroyed(this)).subscribe(() => {
+      this.isFormLoading = false;
+      this.confirmEnableService();
```

Comment:
You could also do `switchMap` here instead of `subscribe` and switch to an observable that would be returned from `confirmEnableService`.